core/src/main/java/org/apache/iceberg/BaseCombinedScanTask.java

```diff
@@ -26,19 +26,19 @@
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 
 public class BaseCombinedScanTask implements CombinedScanTask {
-  private final List<FileScanTask> tasks;
+  private final FileScanTask[] tasks;
 
   public BaseCombinedScanTask(FileScanTask... tasks) {
-    this.tasks = ImmutableList.copyOf(tasks);
+    this.tasks = tasks;
   }
 
   public BaseCombinedScanTask(List<FileScanTask> tasks) {
-    this.tasks = ImmutableList.copyOf(tasks);
+    this.tasks = tasks.stream().toArray(FileScanTask[]::new);
```

Comment:
In Guava, `ImmutableList.copyOf()` will throw on null input. That same behavior isn't mentioned here. Should we document it, either via a `Preconditions` check or via an annotation such as `@ParametersAreNonNullByDefault` which Guava uses.

Note: I can't speak for all versions of Guava obviously (I doubt anybody could at this point), but I found this isue mentioning it https://github.com/google/guava/issues/2470

Reply:
I guess the new code will also throw NPE in any way, but with different error message which is probably less helpful. Checking the null with Precondition sounds good to me, and I prefer defensive programming unless it's on critical path. :) Thanks!